viewer/packages/camera-manager/src/types.ts

```diff
@@ -0,0 +1,89 @@
+/*!
+ * Copyright 2021 Cognite AS
+ */
+
+export type CameraControlsOptions = {
+  /**
+   * Sets mouse wheel initiated action.
+   *
+   * Modes:
+   *
+   * 'zoomToTarget' - zooms just to the current target (center of the screen) of the camera.
+   *
+   * 'zoomPastCursor' - zooms in the direction of the ray coming from camera through cursor screen position, allows going through objects.
+   *
+   * 'zoomToCursor' - mouse wheel scroll zooms towards the point on the model where cursor is hovering over, doesn't allow going through objects.
+   *
+   * Default is 'zoomPastCursor'.
+   *
+   */
+  mouseWheelAction?: 'zoomToTarget' | 'zoomPastCursor' | 'zoomToCursor';
+  /**
+   * Enables or disables change of camera target on mouse click. New target is then set to the point of the model under current cursor position.
+   *
+   * Default is false.
+   *
+   */
+  onClickTargetChange?: boolean;
+};
+
+export type ControlsState = {
+  position: THREE.Vector3;
+  target: THREE.Vector3;
+};
+
+export interface RevealCameraControls extends THREE.EventDispatcher {
```

Comment:
I don't think we need this interface (at least not yet).

I think we should place the abstraction layer at `CameraManager`, and have users implement their own manager instead of drop in controls. This gives more flexibility, but also will require us to add more tooling to enable users to implement this class.

I'm very open for discussion on this topic though...

Reply:
I think I agree about that now. I'll create a possible interface for `CameraManager` and we'll see how it will go. Another thought that I had was creating a base CameraManager class with some "obvious" methods (like fitCameraToModel/BoundingBox) that users can override, so that modification could be easier if you only want to add/change a few methods.

Reply:
Not sure what value this interface adds now - do we want to keep it? My suggestion is to keep the interface for now but dont change the return type in `Cognite3dViewer` and don't expose the interface.

Reply:
Changed to returning `ComboControls` everywhere